In [1]:
import requests
from bs4 import BeautifulSoup

In [6]:
url='https://www.amazon.in/s?k=washing+machine&page=2&crid=QVHWUNQVA5HG&qid=1670153426&sprefix=washing%2Caps%2C396&ref=sr_pg_1'
req=requests.get(url)
print(req)
data=BeautifulSoup(req.content,'html.parser')


<Response [200]>


In [7]:
url_list=[]
for i in range(1,21):
  url_list.append('https://www.amazon.in/s?k=washing&crid=22T9AU7D5S3IJ&qid=1670122845&sprefix=wahsing%2Caps%2C261&ref=sr_pg_' + str(i))
url_list


['https://www.amazon.in/s?k=washing&crid=22T9AU7D5S3IJ&qid=1670122845&sprefix=wahsing%2Caps%2C261&ref=sr_pg_1',
 'https://www.amazon.in/s?k=washing&crid=22T9AU7D5S3IJ&qid=1670122845&sprefix=wahsing%2Caps%2C261&ref=sr_pg_2',
 'https://www.amazon.in/s?k=washing&crid=22T9AU7D5S3IJ&qid=1670122845&sprefix=wahsing%2Caps%2C261&ref=sr_pg_3',
 'https://www.amazon.in/s?k=washing&crid=22T9AU7D5S3IJ&qid=1670122845&sprefix=wahsing%2Caps%2C261&ref=sr_pg_4',
 'https://www.amazon.in/s?k=washing&crid=22T9AU7D5S3IJ&qid=1670122845&sprefix=wahsing%2Caps%2C261&ref=sr_pg_5',
 'https://www.amazon.in/s?k=washing&crid=22T9AU7D5S3IJ&qid=1670122845&sprefix=wahsing%2Caps%2C261&ref=sr_pg_6',
 'https://www.amazon.in/s?k=washing&crid=22T9AU7D5S3IJ&qid=1670122845&sprefix=wahsing%2Caps%2C261&ref=sr_pg_7',
 'https://www.amazon.in/s?k=washing&crid=22T9AU7D5S3IJ&qid=1670122845&sprefix=wahsing%2Caps%2C261&ref=sr_pg_8',
 'https://www.amazon.in/s?k=washing&crid=22T9AU7D5S3IJ&qid=1670122845&sprefix=wahsing%2Caps%2C261&ref=sr

In [8]:
# title_name

title_name=data.find_all('span', class_='a-size-medium a-color-base a-text-normal')

title_name

[<span class="a-size-medium a-color-base a-text-normal">Voltas Beko 7 Kg Semi-Automatic Top Loading Washing Machine (WTT70ALIM, Burgundy Qpaque)</span>,
 <span class="a-size-medium a-color-base a-text-normal">LG 8.0 Kg 5 Star Inverter Touch Control Fully-Automatic Front Load Washing Machine with Heater(FHM1408BDL, Silver, 6 Motion Direct Drive)</span>,
 <span class="a-size-medium a-color-base a-text-normal">LG 8 Kg 5 Star Inverter Fully-Automatic Top Loading Washing Machine (T80SJMB1Z, Middle Black, Jet Spray+)</span>,
 <span class="a-size-medium a-color-base a-text-normal">LG 7 Kg 5 Star Inverter Fully-Automatic Top Loading Washing Machine (T70SJMB1Z, Middle Black, Jet Spray+)</span>,
 <span class="a-size-medium a-color-base a-text-normal">LG 7.5 Kg 5 Star Smart Inverter Fully-Automatic Top Load Washing Machine (T75SKSF1Z, Middle Free Silver, TurboDrum | Smart Motion)</span>,
 <span class="a-size-medium a-color-base a-text-normal">IFB 6.5 Kg 5 Star Front Load Washing Machine 2X Power 

In [9]:
title_name1=[]
for i in title_name:
  title_name1.append(i.text)
  print(title_name1)


['Voltas Beko 7 Kg Semi-Automatic Top Loading Washing Machine (WTT70ALIM, Burgundy Qpaque)']
['Voltas Beko 7 Kg Semi-Automatic Top Loading Washing Machine (WTT70ALIM, Burgundy Qpaque)', 'LG 8.0 Kg 5 Star Inverter Touch Control Fully-Automatic Front Load Washing Machine with Heater(FHM1408BDL, Silver, 6 Motion Direct Drive)']
['Voltas Beko 7 Kg Semi-Automatic Top Loading Washing Machine (WTT70ALIM, Burgundy Qpaque)', 'LG 8.0 Kg 5 Star Inverter Touch Control Fully-Automatic Front Load Washing Machine with Heater(FHM1408BDL, Silver, 6 Motion Direct Drive)', 'LG 8 Kg 5 Star Inverter Fully-Automatic Top Loading Washing Machine (T80SJMB1Z, Middle Black, Jet Spray+)']
['Voltas Beko 7 Kg Semi-Automatic Top Loading Washing Machine (WTT70ALIM, Burgundy Qpaque)', 'LG 8.0 Kg 5 Star Inverter Touch Control Fully-Automatic Front Load Washing Machine with Heater(FHM1408BDL, Silver, 6 Motion Direct Drive)', 'LG 8 Kg 5 Star Inverter Fully-Automatic Top Loading Washing Machine (T80SJMB1Z, Middle Black, J

In [10]:
# rating
p_rating=data.find_all('span', class_='a-icon-alt')
p_rating

[<span class="a-icon-alt">3.8 out of 5 stars</span>,
 <span class="a-icon-alt">4.4 out of 5 stars</span>,
 <span class="a-icon-alt">4.4 out of 5 stars</span>,
 <span class="a-icon-alt">4.5 out of 5 stars</span>,
 <span class="a-icon-alt">4.2 out of 5 stars</span>,
 <span class="a-icon-alt">4.2 out of 5 stars</span>,
 <span class="a-icon-alt">4.4 out of 5 stars</span>,
 <span class="a-icon-alt">4.4 out of 5 stars</span>,
 <span class="a-icon-alt">4.2 out of 5 stars</span>,
 <span class="a-icon-alt">3.3 out of 5 stars</span>,
 <span class="a-icon-alt">4.4 out of 5 stars</span>,
 <span class="a-icon-alt">4.4 out of 5 stars</span>,
 <span class="a-icon-alt">4.4 out of 5 stars</span>,
 <span class="a-icon-alt">4.3 out of 5 stars</span>,
 <span class="a-icon-alt">3.7 out of 5 stars</span>,
 <span class="a-icon-alt">4.7 out of 5 stars</span>,
 <span class="a-icon-alt">4 Stars &amp; Up</span>,
 <span class="a-icon-alt">3 Stars &amp; Up</span>,
 <span class="a-icon-alt">2 Stars &amp; Up</span>,

In [11]:
p_rating1=[]
for i in p_rating:

  p_rating1.append(i.text)
  print(p_rating1)

['3.8 out of 5 stars']
['3.8 out of 5 stars', '4.4 out of 5 stars']
['3.8 out of 5 stars', '4.4 out of 5 stars', '4.4 out of 5 stars']
['3.8 out of 5 stars', '4.4 out of 5 stars', '4.4 out of 5 stars', '4.5 out of 5 stars']
['3.8 out of 5 stars', '4.4 out of 5 stars', '4.4 out of 5 stars', '4.5 out of 5 stars', '4.2 out of 5 stars']
['3.8 out of 5 stars', '4.4 out of 5 stars', '4.4 out of 5 stars', '4.5 out of 5 stars', '4.2 out of 5 stars', '4.2 out of 5 stars']
['3.8 out of 5 stars', '4.4 out of 5 stars', '4.4 out of 5 stars', '4.5 out of 5 stars', '4.2 out of 5 stars', '4.2 out of 5 stars', '4.4 out of 5 stars']
['3.8 out of 5 stars', '4.4 out of 5 stars', '4.4 out of 5 stars', '4.5 out of 5 stars', '4.2 out of 5 stars', '4.2 out of 5 stars', '4.4 out of 5 stars', '4.4 out of 5 stars']
['3.8 out of 5 stars', '4.4 out of 5 stars', '4.4 out of 5 stars', '4.5 out of 5 stars', '4.2 out of 5 stars', '4.2 out of 5 stars', '4.4 out of 5 stars', '4.4 out of 5 stars', '4.2 out of 5 stars']
[

In [12]:
# price
price_value=data.find_all('span', class_='a-price-whole')
price_value1=[]
for i in price_value:
  price_value1.append(i.text)
  print(price_value1)

['9,600']
['9,600', '33,990']
['9,600', '33,990', '22,941']
['9,600', '33,990', '22,941', '21,200']
['9,600', '33,990', '22,941', '21,200', '18,480']
['9,600', '33,990', '22,941', '21,200', '18,480', '29,990']
['9,600', '33,990', '22,941', '21,200', '18,480', '29,990', '13,490']
['9,600', '33,990', '22,941', '21,200', '18,480', '29,990', '13,490', '13,990']
['9,600', '33,990', '22,941', '21,200', '18,480', '29,990', '13,490', '13,990', '36,990']
['9,600', '33,990', '22,941', '21,200', '18,480', '29,990', '13,490', '13,990', '36,990', '21,490']
['9,600', '33,990', '22,941', '21,200', '18,480', '29,990', '13,490', '13,990', '36,990', '21,490', '2,699']
['9,600', '33,990', '22,941', '21,200', '18,480', '29,990', '13,490', '13,990', '36,990', '21,490', '2,699', '23,390']
['9,600', '33,990', '22,941', '21,200', '18,480', '29,990', '13,490', '13,990', '36,990', '21,490', '2,699', '23,390', '23,649']
['9,600', '33,990', '22,941', '21,200', '18,480', '29,990', '13,490', '13,990', '36,990', '21

In [13]:
print(len(title_name1))
print(len(p_rating1))
print(len(price_value1))

16
20
17


In [14]:
title_name1=title_name1[:15]
p_rating1=p_rating1[:15]
price_value1=price_value1[:15]
print(len(title_name1))
print(len(p_rating1))
print(len(price_value1))

15
15
15


In [15]:
import pandas as pd
import numpy as np
df=pd.DataFrame({'Name':title_name1,'Rating':p_rating1,'Price':price_value1})

df

Name              Rating  \
0   Voltas Beko 7 Kg Semi-Automatic Top Loading Wa...  3.8 out of 5 stars   
1   LG 8.0 Kg 5 Star Inverter Touch Control Fully-...  4.4 out of 5 stars   
2   LG 8 Kg 5 Star Inverter Fully-Automatic Top Lo...  4.4 out of 5 stars   
3   LG 7 Kg 5 Star Inverter Fully-Automatic Top Lo...  4.5 out of 5 stars   
4   LG 7.5 Kg 5 Star Smart Inverter Fully-Automati...  4.2 out of 5 stars   
5   IFB 6.5 Kg 5 Star Front Load Washing Machine 2...  4.2 out of 5 stars   
6   LG 7 kg 5 Star Semi-Automatic Top Loading Wash...  4.4 out of 5 stars   
7   LG 9.0 Kg 5 Star Inverter Fully-Automatic Fron...  4.4 out of 5 stars   
8   Whirlpool 7.5 Kg Fully-Automatic Top Loading W...  4.2 out of 5 stars   
9   ElectroSky Portable Handy Washing Machine/Mini...  3.3 out of 5 stars   
10  LG 8.0 Kg Inverter Fully-Automatic Top Loading...  4.4 out of 5 stars   
11  LG 7 kg Inverter Fully-Automatic Top Loading W...  4.4 out of 5 stars   
12  LG 10 Kg 5 Star Smart Inverter Fully-Automatic...  4.4 out of 5 stars   
13  LG 6.5 Kg 5 Star Smart Inverter Fully-Automati...  4.3 out of 5 stars   
14  The Bling Stores Mini Foldable Washing Machine...  3.7 out of 5 stars   

     Price  
0    9,600  
1   33,990  
2   22,941  
3   21,200  
4   18,480  
5   29,990  
6   13,490  
7   13,990  
8   36,990  
9   21,490  
10   2,699  
11  23,390  
12  23,649  
13  20,415  
14  27,499

In [16]:
df.head()

Name              Rating  \
0  Voltas Beko 7 Kg Semi-Automatic Top Loading Wa...  3.8 out of 5 stars   
1  LG 8.0 Kg 5 Star Inverter Touch Control Fully-...  4.4 out of 5 stars   
2  LG 8 Kg 5 Star Inverter Fully-Automatic Top Lo...  4.4 out of 5 stars   
3  LG 7 Kg 5 Star Inverter Fully-Automatic Top Lo...  4.5 out of 5 stars   
4  LG 7.5 Kg 5 Star Smart Inverter Fully-Automati...  4.2 out of 5 stars   

    Price  
0   9,600  
1  33,990  
2  22,941  
3  21,200  
4  18,480

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Name    15 non-null     object
 1   Rating  15 non-null     object
 2   Price   15 non-null     object
dtypes: object(3)
memory usage: 488.0+ bytes


In [38]:
df.to_csv("Washing_machine_details.csv")

**Conclusion:**  The project is given on No broker to scrapp the data of 2BHK&3BHK data but i am not able to fetch data may be that website blocked my ip address so I scrapped washing machine data from Amezon website.

**1**)The highest price of waching machine is Whirlpool 7.5 Kg Fully-Automatic Top Loading Washing Machine and price is 36,990 rupees
**2**)The lowest price of waching machine is LG 8.0 Kg Inverter Fully-Automatic Top Loading Washing Machine and price is 2,699 rupees
**3**)Best rating product is LG 7 Kg 5 Star Inverter Fully-Automatic Top Loading Washing Machine(**4.5 star out of 5**)